### ISSUE TITLE
Markdown extensions only works from blueprint + extending default parser can break control panel

### ISSUE THREAD

**Issue by tao** [statamic/cms#2579]
## Bug Description
This bug was discovered from my control panel breaking... but it's related to the markdown parsers.

I had extended the default markdown parser first in `AppServiceProvider.php`:

```php
$config = [];
Markdown::extend('default', function ($parser) use ($config) {
    return $parser
        ->newInstance($config)
        ->addExtensions(function () {
            return [new FootnoteExtension];
        });
});
```

This worked fine, I had some collections which used footnotes and everything was OK. The problem is I had another collection which used a special modifier called `transcript` which parsed the content a specific way. This included using special types of footnotes that are different from the CommonMark spec.

We call the modified like this:

```html
<article class="w-full md:w-2/3">
        {{ content | raw | transcript }}
</article>
```

When I visited my `transcripts` on the Control Panel I received an `Call to member function appendChild() on null`. This occurred because the default markdown parser tried to read my content and found the badly formatted footnotes.

First issue: Should the control panel scan the entry content on the collections page and try parse it? Just the metadata would be better? Although `content` is a viewable option on the `customise-columns` menu... wouldn't it be better to read this as raw text because it only shows a few words before being truncated? I could change this in the blueprint to no longer be `type: markdown` but I still want the user to have the markdown editor as we're just using a different markdown spec.

After this I tried to create a special markdown parser instead of extending the default one:

```php
$config = [];
Markdown::extend('footnotes', function ($parser) use ($config) {
    return $parser
        ->newInstance($config)
        ->addExtensions(function () {
            return [new FootnoteExtension];
        });
});
```

Second related issue: This only works if you specify the markdown parser in the collection blueprint:

```
      -
        handle: content
        field:
          type: markdown
          parser: footnotes
          display: Content
          localizable: true
``` 

However, if you don't specify this in the blueprint and just call the custom parser from antlers it doesn't work as the [documentation describes](https://statamic.dev/extending/markdown#using-a-custom-parser-in-a-modifier):

```html
<article class="w-full md:w-2/3">
        {{ content | markdown:footnotes }}
</article>
```

## Environment

**Statamic version:** 3.0.7 (but will update to latest release soon and report back it if resolves this issue)

**PHP version:** 7.4.6

**Comment by jasonvarga** [comment]
This was a little hard to follow but I'll do my best here.

The `Call to member function appendChild() on null` error you provided was a little vague. Where'd it come from? We'd need the stack trace from your logs to work this out.

If you plan to have a special markdown syntax on a specific field, you should specify the parser in the blueprint for that field.
Then the control panel will use that in listings, and on the frontend. You don't need to use the markdown modifier if your field is markdown.

**Comment by tao** [comment]
Here's an example of the dynamic parser not working correctly. https://github.com/tao/cool-markdown

If you view the article (/two/dance) with the parser defined in the blueprint it does render correctly...

<img width="1078" alt="Screenshot 2020-10-04 at 13 05 22" src="https://user-images.githubusercontent.com/1446331/95012578-6d856900-0642-11eb-816b-a5f06aff0199.png">

If you view the article (/three/dance) with the dynamic parser `{{ content | markdown:footnotes }}` You'll see it doesn't render correctly...

<img width="1078" alt="Screenshot 2020-10-04 at 13 05 16" src="https://user-images.githubusercontent.com/1446331/95012576-6a8a7880-0642-11eb-96f5-dd39f37270a2.png">

I've added links on the homepage.

The second issue I'm still trying to re-create for you (when I extend the default markdown parser and I believe it's trying to read my custom footnotes in the body content), but this appears in the Control Panel.

<img width="1238" alt="Screenshot 2020-10-04 at 12 49 59" src="https://user-images.githubusercontent.com/1446331/95012653-074d1600-0643-11eb-8a7b-cfbe8046a6e2.png">
<img width="1238" alt="Screenshot 2020-10-04 at 12 50 31" src="https://user-images.githubusercontent.com/1446331/95012657-12a04180-0643-11eb-8d85-cdecee098f21.png">

**Comment by jasonvarga** [comment]
In the github repo you linked, there's no transcripts collection. Is that the only one with the error?

**Comment by tao** [comment]
I tried to create two repositories to show both errors separately. The repo linked previously shows the error with the dynamic markdown parser `{{ content | markdown:footnotes }}`. I haven't been able to recreate the second error yet... perhaps it's best if I create a new issue for that once I isolate it better.

**Comment by tao** [comment]
The `appendChild` error only appears when I extend the default markdown parser but other than that it's been difficult to re-create.
